robbie_viewer_server/utils.py

```diff
@@ -0,0 +1,275 @@
+from functools import lru_cache
+from astropy.io import fits
+from astropy.io.votable import parse
+from astropy.wcs import WCS
+
+import numpy as np
+import datetime
+
+from bokeh.plotting import figure, show, output_file
+from bokeh import palettes
+from bokeh.layouts import gridplot, layout
+from bokeh.io import curdoc
+from bokeh.models import (ColumnDataSource, Circle, Whisker,
+                          DataTable, TableColumn, NumberFormatter,
+                          CustomJS, Slider, DatetimeTickFormatter,
+                          HoverTool)
+TOOLS = "box_select,lasso_select,help,pan,tap,wheel_zoom,reset"
+selected_circle = Circle(fill_alpha=1, fill_color="firebrick", line_color=None)
+nonselected_circle = Circle(fill_alpha=0.2, fill_color="blue", line_color=None)
+
+
+@lru_cache
+def load_mean_image(filename):
+    hdu = fits.open(filename)#'results/mean_image.fits')
+    return hdu
```

Comment:
Is this all the magic you needed to only load the fits files you need?

Reply:
If we're talking about the lazy loading, that hasn't been implemented yet (and doesn't seem trivial to implement). This portion of the code that @PaulHancock wrote is to just load in a mean image .fits but I'm not exactly sure what `@lru_cache` does...

Reply:
The `@lru_cache` will just cache the result of a function call so that it becomes a dictionary look-up the second time you call the function with the same arguments.